tests/test_server.py

```diff
@@ -1232,6 +1229,28 @@ async def test_predict(rasa_app: SanicASGITestClient):
     assert "policy" in content
 
 
+async def test_predict_invalid_entities_format(rasa_app: SanicASGITestClient):
+    data = [
+        {"event": "action", "name": "action_listen"},
+        {
+            "event": "user",
+            "text": "hello",
+            "parse_data": {
+                "entities": {},
+                "intent": {"confidence": 0.57, INTENT_NAME_KEY: "greet"},
+                "text": "hello",
+            },
+        },
+    ]
+
+    _, response = await rasa_app.post(
+        "/model/predict",
+        json=data,
+        headers={"Content-Type": rasa.server.JSON_CONTENT_TYPE},
+    )
+    assert response.status == HTTPStatus.INTERNAL_SERVER_ERROR
```

Comment:
it's not an internal server error but a bad request which leads to that. Hence, I'd use 400